platforms/core-configuration/declarative-dsl-core/src/main/resources/org/gradle/internal/declarativedsl/parsing/grammar.bnf

```diff
@@ -44,22 +44,23 @@
 
 root ::= statements
 
-statement ::= (assignment | expression)
-statements ::= (statement semis?)*
+statement ::= (assignment | block | expression)
+statements ::= (statement semis*)*
 
-assignment ::= directlyAssignableExpression EQ primaryExpression {pin=2}
-expression ::= simpleIdentifier postfixUnarySuffixes
+assignment ::= directlyAssignableExpression EQ primaryExpression (navigationSuffix* valueArguments)? {pin=2}
```

Comment:
It still allows things like `"foo".bar(1)`, right? 
Should we keep it this way or prohibit such calls on anything other than an identifier?

Reply:
Good point, making the change.